x-pack/plugins/triggers_actions_ui/public/application/hooks/use_load_rules.ts

```diff
@@ -114,7 +116,7 @@ export function useLoadRules({
         http,
         page,
         searchText,
-        typesFilter,
+        typesFilter: typesFilter && typesFilter.length > 0 ? typesFilter : filteredRuleTypes,
```

Comment:
@mgiota What is the difference between `typesFilter` and `filteredRuleTypes`? Why do we need to add `filteredRuleTypes`?

Reply:
@maryam-saeidi If we don't add `filteredRuleTypes`, the table would show us rules from all rule types, not only observability rule types. If you look at the initial ticket in the `Implementation details` section https://github.com/elastic/kibana/issues/123580 you can find more details regarding how we can get only o11y rule types. 

`typesFilter` on the otherhand is the filtering user applies using the `Type` filter dropdown.

Does it make sense? I can explain later in more detail. I am open to further improvements if this part doesn't make much sense.